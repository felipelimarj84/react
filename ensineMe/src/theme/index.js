const theme = {
  palette: {
    primary: {
      main: "#7f3f98",
    },
    secondary: {
      main: "#17243c",
      light: "#bbcce9",
    },
    neutral: {
      darkest: "#000000",
      darker: "#333333",
      dark: "#4d4d4d",
      medium: "#808080",
      light: "#bfbfbf",
      lighter: "#f2f2f2",
      lightest: "#ffffff",
    },
    success: {
      dark: "#147c38",
      light: "#e7f2eb",
    },
    error: {
      dark: "#df402b",
      light: "#fcecea",
    },
    alert: {
      dark: "#ddac00",
      light: "#fcf7e5",
    },
    information: {
      dark: "#2f80ed",
      light: "#eaf2fd",
    },
  },
};

export default theme;
