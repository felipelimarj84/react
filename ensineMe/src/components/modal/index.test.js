import React from "react";
import Modal from "./index";
import { ThemeProvider } from "styled-components";
import theme from "../../theme";
import "jest-styled-components";

import { cleanup, render } from "@testing-library/react";

afterEach(cleanup);

it("<Modal/> matches snapshot basic", () => {
  const component = render(
    <ThemeProvider theme={theme}>
      <Modal onClose={() => {}}>
        <h2>Modal do App</h2>
        <p>Isso é um parágrafo</p>
      </Modal>
    </ThemeProvider>
  );

  expect(component.container).toMatchSnapshot();
});
