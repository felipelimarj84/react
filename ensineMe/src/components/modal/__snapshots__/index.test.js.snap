// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal/> matches snapshot basic 1`] = `
.c0 {
  width: 100%;
  height: 100vh;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 10;
  background-color: rgba(0,0,0,0.8);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .container {
  background-color: #fff;
  color: #000;
  width: 366px;
  height: 313px;
  border-radius: 5px;
}

.c0 .content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .close {
  background-color: transparent;
  border: none;
  outline: none;
  width: 32px;
  height: 32px;
  right: calc(-100% + 64px);
  cursor: pointer;
  top: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .close:before,
.c0 .close:after {
  content: "";
  position: absolute;
  width: 2.5px;
  height: 24px;
  background-color: #000;
}

.c0 .close:before {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c0 .close:after {
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

<div>
  <div
    class="c0"
    id="modal"
  >
    <div
      class="container"
    >
      <button
        class="close"
      />
      <div
        class="content"
      >
        <h2>
          Modal do App
        </h2>
        <p>
          Isso é um parágrafo
        </p>
      </div>
    </div>
  </div>
</div>
`;
